pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -228,99 +241,141 @@ func (r *rangeAllocator) AllocateOrOccupyCIDR(node *v1.Node) error {
 		klog.V(2).Infof("Node %v is already in a process of CIDR assignment.", node.Name)
 		return nil
 	}
-	if node.Spec.PodCIDR != "" {
-		return r.occupyCIDR(node)
+
+	if len(node.Spec.PodCIDRs) > 0 {
```

Comment:
Sorry if this was mentioned in the KEP already, are we documenting that a v4 or v6 only cluster can't upgrade to an alpha enabled dual-stack cluster?

Reply:
This also applies in the API validation we do, we only apply both dual-stack addresses if none is set previously but error if 1 is set

Reply:
a `v4 only` or a `v6 only` can be upgraded to a dualstack. The cluster will be functional before and after upgrade. What will not light up is dualstack, because of `don't update cidr if there is a cidr already assigned`. We can document we can light it up.